scikeras/wrappers.py

```diff
@@ -871,192 +874,218 @@ def preprocess_y(y):
             # encode
-            encoders_ = [LabelEncoder() for _ in range(len(y))]
-            y = [
-                encoder.fit_transform(y_.reshape(-1,) if y_.shape[1] == 1 else y_)
-                for encoder, y_ in zip(encoders_, y)
-            ]
-            classes_ = [encoder.classes_ for encoder in encoders_]
+            # here we just pass through to allow all Keras use cases
+            # that didn't exist in sklearn to succeed (ex: loss=categorical_crossentropy)
+            if reset:
+                encoder = FunctionTransformer()
+            else:
+                encoder = self.encoders_[0]
+            y = encoder.fit_transform(y)
+            # there is no real "classes" representation for this type of data
+            classes_ = None
+            n_classes_ = y.shape[1]
+            # make lists
+            encoders_ = [encoder]
+            classes_ = [classes_]
+            n_classes_ = [n_classes_]
         else:
-            raise ValueError("Unknown label type: {}".format(target_type_))
+            raise ValueError(
+                f"Unknown label type: {self.target_type_}."  # To match errors used by sklearn
+                "\n\nTo implement support, subclass KerasClassifier and override"
+                " `preprocess_y` and `postprocess_y`."
+                "\n\nSee (TODO: link to docs) for more information."
+            )
 
-        # self.classes_ is kept as an array when n_outputs>1 for compatibility
+        # self.classes_ is kept as an array when n_outputs==1 for compatibility
         # with ensembles and other meta estimators
         # which do not support multioutput
         if len(classes_) == 1:
-            n_classes_ = classes_[0].size
+            n_classes_ = n_classes_[0]
             classes_ = classes_[0]
             n_outputs_ = 1
-        else:
-            n_classes_ = [class_.shape[0] for class_ in classes_]
-            n_outputs_ = len(n_classes_)
-
-        extra_args.update(
-            {
-                "classes_": classes_,
-                "encoders_": encoders_,
-                "n_outputs_": n_outputs_,
-                "model_n_outputs_": model_n_outputs_,
-                "n_classes_": n_classes_,
-                "target_type_": target_type_,
-            }
-        )
+        # else:  # this branch is unused, but kept as an example for users to extend SciKeras
+        #     n_classes_ = [class_.shape[0] for class_ in classes_]
+        #     n_outputs_ = len(n_classes_)
 
-        return y, extra_args
+        if reset:
+            self.classes_ = (
+                self.user_classes_ if self.user_classes_ is not None else classes_
+            )
+            self.encoders_ = encoders_
+            self.n_outputs_ = n_outputs_
+            self.model_n_outputs_ = model_n_outputs_
+            self.n_classes_ = n_classes_
+        else:
+            check = (
+                (self.classes_, classes_)
+                if isinstance(self.classes_, list)
```

Comment:
Why can `self.classes_` be a list? In the if-statement above on target type, `classes_ = [encoder[1].categories_[0]]` or `classes_ = [None]`. Why is that necessary? Or is this left-over code from multi-input multi-output models?

Reply:
For multi-input/output, it has to be a list (and that's how it's handled in sklearn). I was debating if I should leave it or take it out. I opted to keep it hoping that it would make it easier for users to extend for multi-input/output.

Reply:
I made it single-items for now. We can leave the list-part for example notebooks or docs.